src/utils/streams.js

```diff
@@ -0,0 +1,259 @@
+  });
+};
+
+/**
+ * Pipe the given file provided by --file option to process.stdout.
+ */
+const outputFileFunc = () => {
+  const help = readmeObject.outputFile.join('\n');
+  const fileName = argv.file;
+
+  if (!_checkFileName(fileName, help)) {
+    return;
+  }
+
+  const src = fs.createReadStream(fileName);
+  src.on('error', (err) => {
+    console.log('ERROR. File is not exist.', err.message);
+    console.log(readmeObject.outputFile.join('\n'));
+  });
+
+  src
+      .pipe(process.stdout);
+};
+
+/**
+ * Convert file provided by --file option from csv to json and output data
+ * to process.stdout.
+ * --file option is required for this command.
+ */
+const convertFromFileFunc = () => {
+  const help = readmeObject.convertFromFile.join('\n');
+  const fileName = argv.file;
+  if (!_checkFileName(fileName, help)) {
+    return;
+  }
+
+  if (path.extname(fileName) !== '.csv') {
+    console.log('ERROR. File should have ".csv" extension');
+    console.log(help);
+    return;
+  }
+
+  _convertFileFun(fileName, process.stdout, help);
+};
+
+/**
+ * Convert file provided by --file option from csv to json and output data
+ * to a result file with the same name but json extension.
+ */
+const convertToFileFunc = () => {
+  const help = readmeObject.convertToFile.join('\n');
+  const csvFileName = argv.file;
+  if (!_checkFileName(csvFileName, help)) {
+    return;
+  }
+
+  if (path.extname(csvFileName) !== '.csv') {
+    console.log('ERROR. File should have ".csv" extension');
+    console.log(help);
+    return;
+  }
+
+  const filePathObject = path.parse(csvFileName);
+  filePathObject.base = null;
+  filePathObject.ext = '.json';
+  const jsonFileName = path.format(filePathObject);
+
+  _convertFileFun(csvFileName, fs.createWriteStream(jsonFileName), help);
+};
+
+const _getStdinReadline = () => readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  terminal: false,
+});
+
+const _reverseString = (str) => str.split('').reverse().join('');
+
+const _convertStringToUpperCase = (str) => str.toUpperCase();
+
+const _processStdin = (operation) => {
+  const readLine = _getStdinReadline();
+
+  readLine.on('line', (line) => {
+    console.log(operation(line));
+  });
+};
+
+const _checkFileName = (fileName, help) => {
+  if (!fileName || typeof fileName !== 'string') {
+    console.log('ERROR. File is not provided or filename is invalid.');
+    console.log(help);
+    return false;
+  }
+
+  return true;
+};
+
+const _checkPath = (pathToFiles, help) => {
+  if (!pathToFiles || typeof pathToFiles !== 'string') {
+    console.log('ERROR. Path is not provided or invalid.');
+    console.log(help);
+    return false;
+  }
+
+  return true;
+};
+
+const _convertFileFun = (fileName, writeStream, help) => {
+  const readStream = fs.createReadStream(fileName);
+  readStream.on('error', (err) => {
+    console.log(err.message);
+    console.log(help);
+  });
+
+  readStream
+      .pipe(csv())
+      .pipe(writeStream);
```

Comment:
the file with final result will never be a valid json as it will contain something like:
```
{"a":"Example 1","b":"2","c":"3"}
{"a":"4","b":"5","c":"6"}
```
which are 2 stringified javascript objects written to file one after another which does not  make the whole file a valid json. In order to fix that you need to read the whole csv file into memory and then write it to the file as an array:

```
[
   {"a":"Example 1","b":"2","c":"3"},
   {"a":"4","b":"5","c":"6"}
]
```

Reply:
You are right, thank you. Fixed.